inference-engine/src/mkldnn_plugin/nodes/mkldnn_detection_output_node.cpp

```diff
@@ -125,480 +152,753 @@ void MKLDNNDetectionOutputNode::initSupportedPrimitiveDescriptors() {
+            }
+        }
     }
-    memset(dst_data, 0, dst_data_size);
+}
 
-    int count = 0;
-    for (int n = 0; n < N; ++n) {
-        const float *pconf   = reordered_conf_data + n * _num_priors * _num_classes;
-        const float *pboxes  = decoded_bboxes_data + n*_num_priors*4*_num_loc_classes;
-        const int *pindices  = indices_data + n*_num_classes*_num_priors;
-
-        for (int c = 0; c < _num_classes; ++c) {
-            for (int i = 0; i < detections_data[n*_num_classes + c]; ++i) {
-                int idx = pindices[c*_num_priors + i];
-
-                dst_data[count * DETECTION_SIZE + 0] = static_cast<float>(n);
-                dst_data[count * DETECTION_SIZE + 1] = static_cast<float>(_decrease_label_id ? c-1 : c);
-                dst_data[count * DETECTION_SIZE + 2] = pconf[c*_num_priors + idx];
-
-                float xmin = _share_location ? pboxes[idx*4 + 0] :
-                             pboxes[c*4*_num_priors + idx*4 + 0];
-                float ymin = _share_location ? pboxes[idx*4 + 1] :
-                             pboxes[c*4*_num_priors + idx*4 + 1];
-                float xmax = _share_location ? pboxes[idx*4 + 2] :
-                             pboxes[c*4*_num_priors + idx*4 + 2];
-                float ymax = _share_location ? pboxes[idx*4 + 3] :
-                             pboxes[c*4*_num_priors + idx*4 + 3];
-
-                if (_clip_after_nms) {
-                    xmin = (std::max)(0.0f, (std::min)(1.0f, xmin));
-                    ymin = (std::max)(0.0f, (std::min)(1.0f, ymin));
-                    xmax = (std::max)(0.0f, (std::min)(1.0f, xmax));
-                    ymax = (std::max)(0.0f, (std::min)(1.0f, ymax));
+inline void MKLDNNDetectionOutputNode::confReorderDense(const float *confData, const float *ARMConfData, float *reorderedConfData) {
+    if (withAddBoxPred) {
+        parallel_for2d(imgNum, priorsNum, [&](size_t n, size_t p) {
+            if (ARMConfData[n * priorsNum * 2 + p * 2 + 1] < objScore) {
+                for (int c = 0; c < classesNum; ++c) {
+                    reorderedConfData[n * priorsNum * classesNum + c * priorsNum + p] = c == backgroundClassId ? 1.0f : 0.0f;
+                }
+            } else {
+                for (int c = 0; c < classesNum; ++c) {
+                    reorderedConfData[n * priorsNum * classesNum + c * priorsNum + p] = confData[n * priorsNum * classesNum + p * classesNum + c];
                 }
-
-                dst_data[count * DETECTION_SIZE + 3] = xmin;
-                dst_data[count * DETECTION_SIZE + 4] = ymin;
-                dst_data[count * DETECTION_SIZE + 5] = xmax;
-                dst_data[count * DETECTION_SIZE + 6] = ymax;
-
-                ++count;
             }
-        }
+        });
+        return;
     }
-
-    if (count < num_results) {
-        // marker at end of boxes list
-        dst_data[count * DETECTION_SIZE + 0] = -1;
+    // withAddBoxPred is false
+    if (permuteKernel_) {
```

Comment:
Just don't see have the condition can be false. `permuteKernel_` is initialized inside createPrimitive w/o  any condition.

Reply:
Added conditions in createPrimitive(). Thanks!